iOS/DuckDuckGo/Subscription/SubscriptionURLNavigationHandler.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  SubscriptionURLNavigationHandler.swift
+//  DuckDuckGo
+//
+//  Copyright © 2025 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Core
+import BrowserServicesKit
+import Subscription
+
+final class SubscriptionURLNavigationHandler: SubscriptionUserScriptNavigationDelegate {
```

Comment:
Can we add documentation for this class. Also, should we just make the class `@MainActor` rather than annotate each method?